File: 2022/1/python/1.2.py
def read_input() -> list:
    fileContent = ""
    with open("./puzzleInput.txt") as file:
        fileContent = file.read()
    return fileContent.split("\n")

def main():
    totalCalorieList = []
    puzzleInput = read_input()

    currentTotal = 0
    for calories in puzzleInput:
        if not calories:
            totalCalorieList.append(currentTotal)
            currentTotal = 0
            continue
        calorieValue = int(calories)
        currentTotal = currentTotal + calorieValue

    topThreeTotal = 0
    totalCalorieList.sort()
    for total in totalCalorieList[-3:]:
        topThreeTotal = total + topThreeTotal
    
    print(topThreeTotal)


if __name__ == "__main__":
    main()





        

File: 2022/1/go/1.2.go
package main

import (
	"fmt"
	"io/ioutil"
	"sort"
	"strconv"
	"strings"
)

func main() {

	fileContent, err := ioutil.ReadFile("puzzleInput.txt")

	if err != nil {
		fmt.Println(err)
	}

	allItems := strings.Split(string(fileContent), "\r\n\r\n")

	allTotals := []int{}
	for _, e := range allItems {
		currentTotal := 0
		backpack := strings.Split(e, "\r\n")
		for _, f := range backpack {
			n, _ := strconv.Atoi(f)
			currentTotal = currentTotal + n
		}

		allTotals = append(allTotals, currentTotal)
	}

	sort.Ints(allTotals)

	var result int
	for _, t := range allTotals[len(allTotals)-3:] {
		result = result + t
	}

	fmt.Println(result)
}
